File: UnityFiles/FPSTest2/Assets/Scripts/shootAction.js
#pragma strict
var projectile : Rigidbody; 
var speed = 20; 
function Start(){

}
function Update() { 
	if( Input.GetButtonDown( "Fire1" ) ) { 
		var instantiatedProjectile : Rigidbody = Instantiate( projectile, transform.position, transform.rotation );
		instantiatedProjectile.velocity = transform.TransformDirection( Vector3( 0, 0, speed ) );
		Physics.IgnoreCollision( instantiatedProjectile. GetComponent.<Collider>(), transform.GetComponent.<Collider>() );
		Physics.IgnoreCollision( instantiatedProjectile. GetComponent.<Collider>(), transform.root.GetComponent.<Collider>() );
	//	Destory(instantiatedProjectile,2.0f); 
	} 
}

File: UnityFiles/FPSTest2/Assets/Scripts/shootAction.cs
using UnityEngine;
using System.Collections;

public class shootAction : MonoBehaviour {
	public Rigidbody projectile ;
	public float speed = 5.0f;

	// Use this for initialization
	void Start () {
		
	}
	// Update is called once per frame
	void Update () {
		if(Input.GetMouseButtonDown(0)){
			Rigidbody instantiatedProjectile = Instantiate (projectile,transform.position,transform.rotation) as Rigidbody;
			instantiatedProjectile.velocity = transform.TransformDirection (new Vector3(0f,0f,speed));
			Physics.IgnoreCollision ( instantiatedProjectile.GetComponent<Collider>() , transform.root.GetComponent<Collider>());
			Physics.IgnoreCollision ( instantiatedProjectile.GetComponent<Collider>() , transform.GetComponent<Collider>());

		}
	}
}
